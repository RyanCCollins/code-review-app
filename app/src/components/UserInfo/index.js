import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import AvatarMissing from './missing_large.png';

const UserInfo = ({
  user,
}) => (
  <div className={styles.userInfo}>
    <img
      src={user.avatar ? user.avatar : AvatarMissing}
      className={styles.avatarImg}
      alt={`${user.name} avatar`}
    />
    <div className={styles.userInfo}>
      {user.name}
    </div>
  </div>
);

UserInfo.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    avatar: PropTypes.string,
  }),
};

export default cssModules(UserInfo, styles);
