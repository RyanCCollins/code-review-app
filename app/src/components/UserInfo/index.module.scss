.avatarImg {
  display: block;
  width: 40px;
  height: 40px;
  background: #ECEFF1;
  -webkit-border-radius: 100%;
  -moz-border-radius: 100%;
  border-radius: 100%;
  overflow: hidden;
  position: relative;
}

.userInfo {
  display: flex;
  flex-direction: row;
}
