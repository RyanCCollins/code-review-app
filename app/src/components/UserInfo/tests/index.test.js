import UserInfo from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<UserInfo />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <UserInfo
        user={{
          name: 'Ryan Collins',
          avatar: 'https://ryancollins.io/avatar.png',
        }}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
