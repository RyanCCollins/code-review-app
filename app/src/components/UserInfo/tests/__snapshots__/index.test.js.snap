exports[`<UserInfo /> should render with default props 1`] = `
<div>
  <img
    alt="Ryan Collins avatar"
    src="https://ryancollins.io/avatar.png" />
  <div>
    Ryan Collins
  </div>
</div>
`;
